tools/oeedger8r/Emitter.ml

```diff
@@ -379,9 +399,10 @@ let oe_prepare_input_buffer (os:out_channel) (fd:Ast.func_decl) (alloc_func:stri
       | Ast.PTPtr (atype, ptr_attr) ->
         if ptr_attr.Ast.pa_chkptr then
           let size = oe_get_param_size (ptype, decl, "_args.") in
+          let tystr = get_parameter_type_str ptype decl in
```

Comment:
Bah all this code is called so many times and it's sooo similar.

Reply:
Feel free to refactor :)